### PR TITLE
oauth-apiserver: don't attempt to write to system trust store

### PR DESCRIPTION
**Pull request by stlaz**
We don't use the global trusted CA bundle in this API server, don't attempt to write it.

/assign @p0lyn0mial

### CODE REVIEW

File: pkg/operator/assets/bindata.go [hunk 1]
```diff
@@ -329,26 +329,19 @@ spec:
         terminationMessagePolicy: FallbackToLogsOnError
         image: ${IMAGE}
         imagePullPolicy: IfNotPresent
-        command: ["/bin/bash", "-ec"]
+        command: ["oauth-apiserver", "start"]
         args:
-          - |
-            if [ -s /var/run/configmaps/trusted-ca-bundle/tls-ca-bundle.pem ]; then
```
Comment (p0lyn0mial) [inline pkg/operator/assets/bindata.go:335]: it was taken from `oas` - https://github.com/openshift/cluster-openshift-apiserver-operator/blob/master/bindata/v3.11.0/openshift-apiserver/deploy.yaml#L67

should we update there as well?

**Comment by p0lyn0mial** [comment]
> We don't use the global trusted CA bundle in this API server, don't attempt to write it.

we use the CA from the config map, right?

**Comment by p0lyn0mial** [comment]
if this is what we want then we should remove the wiring like https://github.com/openshift/cluster-authentication-operator/blob/master/pkg/operator/starter.go#L395

File: pkg/operator/assets/bindata.go (hunk 1 above)
Comment (stlaz) [reply pkg/operator/assets/bindata.go:335]: no, I think OAS is actually using this, but we're not

**Comment by stlaz** [comment]
> we use the CA from the config map, right?

no, we're the trust roots vendored inside the image

> if this is what we want then we should remove the wiring like https://github.com/openshift/cluster-authentication-operator/blob/master/pkg/operator/starter.go#L395

we can't, that's a different workload that's actually using it :)

**Pull request closed without merging**